sdk/packages/selector-polkadot/src/adapter.ts

```diff
@@ -655,6 +665,26 @@ export class NightlyConnectAdapter
     return this.walletsList
   }
 
+  checkForArrivingWallets = (metadataWallets: WalletMetadata[]) => {
+    clearInterval(this._detectionIntervalId)
+    let checks = 0
+
+    this._detectionIntervalId = setInterval(() => {
+      if (checks >= this._maxNumberOfChecks || this.connected) {
+        clearInterval(this._detectionIntervalId)
+      }
+      checks++
+      this.walletsList = getPolkadotWalletsList(
+        metadataWallets,
+        getRecentWalletForNetwork(this.network)?.walletName ?? undefined
+      )
+    }, 1000)
+  }
+
+  stopIntervalsOnExit = () => {
+    clearInterval(this._detectionIntervalId)
+  }
```

Comment:
what is purpose of this ?

Reply:
to clear the interval from the outside, here it is used to stop the interval when exiting the page